WorldHopper: Add Favorite Star

ConnorVG opened the pull request:
Add a favorite star to the "World Hopper" plugin.

<details>
<summary>How it looks</summary>

![World Hopper](https://user-images.githubusercontent.com/3260091/44051452-d0548e64-9f31-11e8-9eed-647e7fd990e9.png)

</details>

deathbeam commented on runelite-client/src/main/java/net/runelite/client/plugins/worldhopper/WorldTableRow.java:
```diff
@@ -65,14 +67,17 @@
 		FLAG_UK = new ImageIcon(ImageUtil.getResourceStreamFromClass(WorldHopperPlugin.class, "flag_uk.png"));
 		FLAG_US = new ImageIcon(ImageUtil.getResourceStreamFromClass(WorldHopperPlugin.class, "flag_us.png"));
 		FLAG_GER = new ImageIcon(ImageUtil.getResourceStreamFromClass(WorldHopperPlugin.class, "flag_ger.png"));
+		ICON_FAVORITE = new ImageIcon(ImageUtil.getResourceStreamFromClass(WorldHopperPlugin.class, "favorite.png"));
```
Use the in-game star sprite probably. Or reuse the config panel one and move it to common location

ConnorVG replied:
This is the ingame star sprite.

deathbeam replied:
No, I mean from game. E.g SpriteManager.getSprite

ConnorVG replied:
Ah, I get you. Any particular reason we haven't done so for the flags? I am imagining preference of style.

deathbeam replied:
I think @psikoi said the in-game flags are ugly

ConnorVG replied:
Fair enough, just had a quick look and I completely agree 😂

Adam- commented:
Maybe I would put this on the far left. We plan on putting ping on the right.

Kamielvf commented:
Could make it so that it is silver when non-favorite and when clicked it favorites that world.

ConnorVG commented:
@Adam- yeah, I sort of agree with the put it on the left thing I just couldn't think of a tidy way to do that.
@Kamielvf it gets really busy when you do that, but we could just replicate the config's style with it (though, like I said, it gets busy - mainly due to the other icon and other coloured data in the same row).

Nightfirecat commented:
Are you able to use SpriteManager to get this sprite instead of adding an image file to the repo?

ConnorVG commented:
@Nightfirecat I think it makes the most sense, and looks the best, if we use the config's star.

ConnorVG commented:
I've changed the star to be functional (click to toggle favorite / non-favorite) and moved it to the left side.

<details>
<summary>How it looks</summary>

![untitled](https://user-images.githubusercontent.com/3260091/44054169-ce8c9722-9f39-11e8-8b1e-a20c1c04697a.png)

</details>

Nightfirecat commented on runelite-client/src/main/java/net/runelite/client/plugins/worldhopper/WorldTableRow.java:
```diff
@@ -65,14 +71,24 @@
+		final BufferedImage starOnImage = ImageUtil.getResourceStreamFromClass(IconTextField.class, "star.png");
+
+		ICON_FAVORITE_ON = new ImageIcon(starOnImage);
+		ICON_FAVORITE_ON_HOVER = new ImageIcon(ImageUtil.grayscaleOffset(starOnImage, -100));
```
Generally speaking, `alphaOffset` should be used in place of `grayscaleOffset` for hovered images.

Kamielvf commented:
IMO it looks better with the star being everywhere, so the table columns are aligned properly. Also the added UX is nice.

ConnorVG commented:
@Kamielvf yeah, after making it basically mock the plugin config panel... I think I agree.
